### PR TITLE
Fix game of life example on macos

### PR DESCRIPTION
**Pull request by hakolao**
1. [x] Describe in common words what is the purpose of this change, related
   Github Issues, and highlight important implementation aspects.

3. [x] Run `cargo fmt` on the changes.

I noticed that on my macos the `create_vk_surface_from_handle` encounters an error:

```
[mvk-error] VK_ERROR_INITIALIZATION_FAILED: vkCreateMacOSSurfaceMVK(): On-screen rendering requires a layer of type CAMetalLayer.
```
What is done in `winit_to_surface` prevents this. I wasn't even sure why the handle version was used anyway...

### CODE REVIEW

**Comment by Rua** [comment]
I'm not on MacOS so I can't test this, but it seems like a simple enough fix. Can you explain why this fixes it though, and is it maybe something that should be documented in these two functions?

**Comment by Eliah-Lakhin** [comment]
@hakolao Can you address @Rua's question, please?

**Comment by hakolao** [comment]
Apologies for the delay! @Eliah-Lakhin @Rua 

`CAMetalLayer` is the native rendering surface of Apple’s Metal framework. It is required to be able to render on mac. It looks like that the `winit_to_surface` function ensures this layer is set to the `ns_view` before passing it to `Surface::from_mac_os`. 

Without that, the direct `create_surface_from_handle` fails to above error. 

Some references: https://github.com/KhronosGroup/MoltenVK/issues/78
https://www.khronos.org/registry/vulkan/specs/1.3-extensions/man/html/VK_EXT_metal_surface.html

This could be documented somewhere, however I'd think that having it searchable on github might also be enough, and those who use or should use `create_surface_from_handle` are probably very few. I thought I needed it, but turned out I did not...

**Comment by Rua** [comment]
I think this should be included as documentation on `create_surface_from_handle` so that users know they may need to do this manually. It's not very user friendly to ask them to search on Github.

**Comment by hakolao** [comment]
Ok, I’ll add it there!

**Comment by hakolao** [comment]
@Rua would this be better now? I extracted the layer setting to its own function, so if someone wants to use the `from_handle` they can then call that function also on macos.

File: vulkano-win/src/raw_window_handle.rs [hunk 1]
```diff
@@ -5,7 +5,10 @@ use vulkano::swapchain::Surface;
 use vulkano::swapchain::SurfaceCreationError;
 
 /// Creates a vulkan surface from a generic window
-/// which implements HasRawWindowHandle and thus can reveal the os-dependent handle
+/// which implements HasRawWindowHandle and thus can reveal the os-dependent handle.
+/// - Note that if you wish to use this function with MacOS, you will need to ensure that the
+/// `CAMetalLayer` is set to the ns_view. You can do that by calling `vulkano_win::set_ca_metal_layer_to_winit`
```
[the remark this answers is not in the document]
Comment (hakolao) [reply vulkano-win/src/raw_window_handle.rs]: Makes sense. I updated once more...

[the remark this answers is not in the document]
Comment (hakolao) [reply vulkano-win/src/raw_window_handle.rs]: Although... One could if they want to retain the ownership of the window else where? Though that's probably not necessary...

**Pull request merged**